### PR TITLE
Sys_task

### PR DESCRIPTION
**Pull request by PikalaxALT**
Depends on #285

### CODE REVIEW

File: src/battle/overlay_12_0226BEC4.c [hunk 1]
```diff
@@ -17,7 +17,7 @@ SysTask *ov12_0226BEC4(BattleSystem *bsys) {
 }
 
 void ov12_0226BEF0(SysTask *task) {
-    FreeToHeap(sub_0201F988(task));
+    FreeToHeap(SysTask_GetData(task));
```
Comment (adrienntindall) [inline src/battle/overlay_12_0226BEC4.c:20]: 'GetData' feels like a bad name but idk if there's enough information to expand on it further

File: src/sys_task.c [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    return num * (sizeof(SysTask) + sizeof(SysTask *)) + sizeof(SysTaskQueue);
+}
+
+SysTaskQueue *SysTaskQueue_PlacementNew(u32 num, void *p_mem) {
+    GF_ASSERT(p_mem != NULL);
+
+    SysTaskQueue *ret = (SysTaskQueue *)p_mem;
+    SysTask **taskStack = (SysTask **)((char *)p_mem + sizeof(SysTaskQueue));
+    ret->taskStack = taskStack;
+    SysTask *taskList = (SysTask *)((char *)taskStack + num * sizeof(SysTask *));
```
Comment (adrienntindall) [inline src/sys_task.c:60]: can you rewrite this in terms of arrays ie taskStack[num]

File: src/sys_task.c [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+u32 SysTaskQueue_GetArenaSize(u32 num) {
+    return num * (sizeof(SysTask) + sizeof(SysTask *)) + sizeof(SysTaskQueue);
+}
+
+SysTaskQueue *SysTaskQueue_PlacementNew(u32 num, void *p_mem) {
+    GF_ASSERT(p_mem != NULL);
+
+    SysTaskQueue *ret = (SysTaskQueue *)p_mem;
+    SysTask **taskStack = (SysTask **)((char *)p_mem + sizeof(SysTaskQueue));
```
Comment (adrienntindall) [inline src/sys_task.c:58]: same as below

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: src/battle/overlay_12_0226BEC4.c (hunk 1 above)
Comment (PikalaxALT) [reply src/battle/overlay_12_0226BEC4.c:20]: It's just the pointer to the task's stored data (passed to SysTask_CreateOnXYZQueue), which can be literally anything.

File: src/sys_task.c (hunk 2 above)
Comment (PikalaxALT) [reply src/sys_task.c:60]: well whaddaya know, it still matches this way

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**